src/scheduler/workers/analytics/config.ts

```diff
@@ -0,0 +1,32 @@
+import { logger } from "scheduler/logger";
+
+function getFrequencyInDays() {
+  const { ANALYTICS_FREQUENCY_IN_DAYS = 7 } = process.env;
+  if (Number.isNaN(ANALYTICS_FREQUENCY_IN_DAYS)) {
+    logger.warn(
+      `ANALYTICS_FREQUENCY_IN_DAYS is set to ${ANALYTICS_FREQUENCY_IN_DAYS}, it is not a number, using 7 instead`
+    );
+    return 7;
+  }
```

Comment:
for k8s yaml, you can't actually pass in numbers. it has to be a string. so the isNaN check is more to validate that `"100"` can be parsed as `100`. I don't think someone would actually set it to "abcd" for example. 

I don't think `Number` throws unless you pass in something huge (big int), it will just return `NaN`, so I can't use try catch. 

This covers the case that someone passes in `""` - which is a defined string so default value  ` = 7`  won't work.